modules/audit-enqueued-assets/load.php

```diff
@@ -0,0 +1,312 @@
+<?php
+/**
+ * Module Name: Audit Enqueued Assets
+ * Description: Adds a CSS and JS resource checker in Site Health checks.
+ * Focus: site-health
+ * Experimental: No
+ *
+ * @package performance-lab
+ * @since 1.0.0
+ */
+
+/**
+ * Audit enqueued scripts in the frontend. Ignore /wp-includes scripts.
+ *
+ * It will save information in a transient for 12 hours.
+ *
+ * @since 1.0.0
+ */
+function perflab_aea_audit_enqueued_scripts() {
+	if ( ! is_admin() && ! get_transient( 'aea_enqueued_scripts' ) ) {
```

Comment:
Three things about the conditions here:
* It needs to use a stricter check for the transient, since the transient could contain an empty array - it's unlikely that no scripts are enqueued, but it's certainly possible, so we need to change this to a `false === get_transient( 'aea_enqueued_scripts' )` instead.
* More importantly though, this code can set a transient when no user is logged in, which is problematic from both a security and frontend caching perspective. I think we will need to change this so that it only runs when a user is logged in, potentially even only when a user with a certain capability is logged in.
* Last but not least, right now this will record the enqueued scripts data from a "random" page, whatever a user accesses when the transient has expired. On different types of pages, different scripts may be enqueued. I understand we can't reasonably cover all the different scenarios, and especially for the first one we should keep it simple, but I strongly suggest to always use the same context to gather this data, otherwise it would be inconsistent and not very meaningful when comparing a before/after. I suggest we go with `is_front_page()`, i.e. only collecting data from the home page of the site.

I suggest we go with something like the following:
```suggestion
	if ( ! is_admin() && is_front_page() && current_user_can( 'view_site_health_checks' ) && false === get_transient( 'aea_enqueued_scripts' ) ) {
```

Reply:
@felixarntz I agree with the extra checks you propose. Let's go for **is_front_page** for simplicity (for now) maybe we can even update the messaging to mention this? I would say in future iterations, we can capture information on other important pages, like **is_shop** for woo (if installed) and others, since that will be also really valuable info.

Reply:
I've implemented this approach.